### ISSUE TITLE
[false-negative]HE_EQUALS_USE_HASHCODE can not detect equals method overide in method

### ISSUE THREAD

**Issue by wuchiuwong** [spotbugs/spotbugs#1142]
**Affects SpotBugs Version:** 
spotbugs-4.0.0-RC2

**Rule:**
HE_EQUALS_USE_HASHCODE

**Description:**
Unable to detect equals method overide in method
This rule is implemented in SpotBugs.jar\main\java\edu\umd\cs\findbugs\detect\FindHEmismatch.java
The undetectable reason may be that this rule only detects the situation of the class(<code>visitAfter(JavaClass obj)</code>), and does not detect Override in the method

**Code Sample demonstrating the issue:**
```
void sortFields (Field [] fields)
{
	Arrays.sort (fields, new Comparator<Field> () {
		public int compare (Field f1, Field f2)
		{
			return f1.getName ().compareTo (f2.getName ());
		}
		public boolean equals (Object obj)
		{
			return obj == this;
		}
	});
}
```

**Expected outcome:**
false-negative

**Running SpotBugs through:**
command line
